text/batch-l2.md

```diff
@@ -0,0 +1,1540 @@
+
+```ts
+jobDefn.addVolume({
+  name: 'myVolume',
+  efsVolumeConfiguration: {
+    fileSystem: myFileSystem
+  },
+  containerPath: '/Volumes/myVolume',
+});
+```
+
+### Running Kubernetes Workflows:
+
+Batch also supports running workflows on EKS. The following example creates a `JobDefinition` that runs on EKS:
+
+```ts
+const eksContainer = new batch.EksContainerDefinition(this, 'myEksContainer', {
+   image: ContainerImage.fromRegistry('my-registry/my-image:latest'),
+});
+
+const eksPod = new batch.EksPod(this, 'myEksPod', {
+   containers: [eksContainer],
+});
+
+new batch.EksJobDefinition(this, 'myEksJobDefn', {
+   pod: eksPod,
+});
+
+// alternative using convienience methods:
+const pod = new batch.EksPod(this, 'myEksPod');
+const jobDefn = new batch.EksJobDefinition(this, 'myEksJobDefn', {
+  pod,
+});
+
+jobDefn.addContainer('myEksContainer', { // adds to the Pod
+  image: ecs.ContainerImage.fromRegistry('my-registry/my-image:latest'),
+});
+```
+
+You can mount `Volume`s to these containers in a single operation:
+
+```ts
+jobDefn.addEmptyDirVolume({
+  name: 'emptyDir',
+  mountPath: '/Volumes/emptyDir',
+});
+jobDefn.addHostPathVolume({
+  name: 'hostPath',
+  hostPath: '/sys', 
+  mountPath: '/Volumes/hostPath',
+});
+jobDefn.addSecretVolume({
+  name: 'secret', 
+  secret: new Secret(this, 'mySecret'),
+  mountPath: '/Volumes/secret',
+});
+```
+
+Alternatively, if you prefer to add Volumes to the container without mounting them: //do we need this section? I've only implemented this to keep API-parity with our existing ECS construct>
+
+```ts
+// Creates the volumes on the Pod
+jobDefn.addEmptyDirVolume('emptyDir');
+jobDefn.addHostPathVolume('hostPath', '/sys');
+jobDefn.addSecretVolume('secret', new Secret(this, 'mySecret'));
+```
+
+You can mount these later:
+
+```ts
+// Mounts them so they are accessible to the containers
+jobDefn.addVolumeMount('emptyDir', '/Volumes/emptyDir');
+jobDefn.addVolumeMount('hostPath', '/Volumes/hostPath');
+jobDefn.addVolumeMount('secret', '/Volumes/secret');
+```
+
+### Running Distributed Workflows
+
+Some workflows benefit from parallellization and are most powerful when run in a distributed environment,
+such as certain numerical calculations or simulations. Batch offers `MultiNodeJobDefinition`s for this purpose.
+Message Passing Interface (MPI) is often used with these workflows.
+You must configure your containers to use MPI properly,
+but Batch allows different nodes running different containers to communicate easily with one another.
+You must configure your containers to use certain environment variables that Batch will provide them,
+which lets them know which one is the main node, among other information.
+In particular, the environment variable that tells the containers which one is the main node can be configured on your `MultiNodeJobDefinition` as follows:
+
+```ts
+const multiNodeJob = new batch.MultiNodeJobDefinition(this, 'JobDefinition', {
+  mainNode: 0,
```

Comment:
Please make 0 implicit though, right?

Reply:
Will the total amount of instances automatically be `max(endNode) - min(startNode) + 1` ? Or will we always start counting from 0, even if it doesn't run any code?

Reply:
we always start from 0, but the total amount of instances is always `endNode + 1`. So if we have 4 nodes, we must define node 0, node 1, node 2, and node 3.